source/fe/fe_values.cc

```diff
@@ -3479,9 +3482,9 @@ maybe_invalidate_previous_present_cell (const typename Triangulation<dim,spacedi
           // connect to the current one; also invalidate the previous
           // cell because we shouldn't be comparing cells from different
           // triangulations
-          tria_listener.disconnect ();
+          tria_listener[0].disconnect ();
           invalidate_present_cell();
```

Comment:
Yes, one needs to treat both signals, but that is done in `invalidate_present_cell` already.